pydis_site/apps/api/viewsets/bot/mailing_list.py

```diff
@@ -0,0 +1,85 @@
+from django.db import IntegrityError
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.exceptions import ParseError
+from rest_framework.mixins import ListModelMixin, RetrieveModelMixin
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.viewsets import GenericViewSet
+
+from pydis_site.apps.api.serializers import MailingListSerializer
+from pydis_site.apps.api.models import MailingList, MailingListSeenItem
+
+
+class MailingListViewSet(GenericViewSet, ListModelMixin, RetrieveModelMixin):
```

Comment:
As I was making the bot side, I realised that the fact that we can't create mailing lists is an issue for the following reasons:
1. We don't have any migrations that create theses mailing lists, so if you delete a container that used to have th old table structure, you'll never end up with the target mailing lists.
2. We used to workaround this bot side thanks to the `sync_mailing_lists` method that creates the proper JSON that will be dumped in the db, this is no longer true
3. The bot will keep on getting `404` as it has as hardcoded mailing list that it tracks, so when it tries to post to site, it will complain saying it can't find such registered list.


We should use `ListCreateModelMixin` instead to allow the bot to create the mailing lists.

Reply:
Do you mean `CreateModelMixin`? I can't find anything about `ListCreateModelMixin`.

Reply:
I always get confused, I was talking about the ApiView.

Yes you're right, it's the CreateModelMixin

Reply:
Pushed this.